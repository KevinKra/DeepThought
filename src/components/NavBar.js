import React from "react";
import "../styles/_NavBar.scss";
class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  SignOut = e => {
    e.preventDefault();
    this.props.history.push(`/`);
  };
  toProfile = e => {
    e.preventDefault();
  };
  toMain = e => {
    e.preventDefault();
    this.props.history.push(`/main/1`);
  };
  renderMain = () => {
    return (
      <button
        className="btn"
        history={this.props.history}
        onClick={this.toMain}
      >
        <p href="">Main Page</p>
      </button>
    );
  };
  render() {
    return (
      <header className="navBar">
        <h1 className="mainHeader">ThinkDeeper</h1>
        <form className="navLinks">
          <button
            className="btn"
            history={this.props.history}
            onClick={this.toProfile}
          >
            <p href="">Profile</p>
          </button>
          {this.props.mainLink ? this.renderMain() : null}
          <button
            className="btn"
            history={this.props.history}
            onClick={this.SignOut}
          >
            <p>Sign Out</p>
          </button>
        </form>
      </header>
    );
  }
}

export default NavBar;
