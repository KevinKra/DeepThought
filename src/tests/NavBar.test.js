import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import NavBar from "../components/NavBar";

const defaultState = {};

describe("<NavBar />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<NavBar />);
  });
  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should match default state", () => {
    expect(wrapper.state()).toEqual(defaultState);
  });
});
