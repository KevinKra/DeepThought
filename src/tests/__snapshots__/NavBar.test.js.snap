// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match the snapshot 1`] = `
<header
  className="navBar"
>
  <h1
    className="mainHeader"
  >
    ThinkDeeper
  </h1>
  <form
    className="navLinks"
  >
    <button
      className="btn"
    >
      <p>
        Profile
      </p>
    </button>
    <button
      className="btn"
    >
      <p>
        Sign Out
      </p>
    </button>
  </form>
</header>
`;
